src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -72,6 +77,55 @@ def get_default_ophys_metadata():
     return metadata
 
 
+def get_nwb_imaging_metadata(imgextractor: ImagingExtractor):
+    """
+    Convert metadata from the ImagingExtractor into nwb specific metadata.
+
+    Parameters
+    ----------
+    imgextractor: ImagingExtractor
+    """
+    metadata = get_default_ophys_metadata()
+    # Optical Channel name:
+    channel_name_list = imgextractor.get_channel_names()
+    if channel_name_list is None:
+        channel_name_list = ["generic_name"] * imgextractor.get_num_channels()
+
+    for index, channel_name in enumerate(channel_name_list):
+        if index == 0:
+            metadata["Ophys"]["ImagingPlane"][0]["optical_channel"][index]["name"] = channel_name
+        else:
+            metadata["Ophys"]["ImagingPlane"][0]["optical_channel"].append(
+                dict(
+                    name=channel_name,
+                    emission_lambda=np.nan,
+                    description=f"{channel_name} description",
+                )
+            )
+    # set imaging plane rate:
+    rate = (
+        np.float("NaN")
+        if imgextractor.get_sampling_frequency() is None
```

Comment:
Can this actually ever be `None` currently? If so that's a problem, it's more expected (as per our previous conversations) that this be a required value for all extractors.

Reply:
If so, then we can get rid of this.

Reply:
@CodyCBakerPhD OK, in fact, the class `Hdf5ImagingExtractor(ImagingExtractor)` does not require `sampling_frequency` in its required values and we have it like that in our tests.

Reply:
Dang, we should probably either fix that, or it's 'optional' because it might be an attribute associated with the `Dataset` object (but required if the `Dataset` doesn't have one).